### PR TITLE
consolidate arm core specific constants

### PR DESCRIPTION
**Pull request by axel-h**

### CODE REVIEW

File: libsel4/sel4_plat_include/exynos5/sel4/plat/api/constants.h [hunk 1]
```diff
@@ -7,14 +7,20 @@
...
+/* This is used for multiple SOCs:
+ * - Exynos5250 (A15)
+ * - Exynos5410 (A7/A15)
+ * - Exynos5422 (A7/A15)
+ * Since the A7 was designed as big.LITTLE companion for the A15, their settings
+ * are identical.
+ */
+#if defined(CONFIG_ARM_CORTEX_A7)
+#include <sel4/arch/constants-cortex-a7.h>
+#el#if defined(CONFIG_ARM_CORTEX_A15)
```
Comment (axel-h) [inline libsel4/sel4_plat_include/exynos5/sel4/plat/api/constants.h:19]: need to fix this
```suggestion
#elif defined(CONFIG_ARM_CORTEX_A15)
```

**Comment by axel-h** [comment]
anybody has an idea why `TX2_debug_gcc_64 FAILED` failes?

**Review by kent-mcleod — CHANGES_REQUESTED** [review]
Thanks for this!  Are you able to use `_` instead of `-` in the new filenames to be consistent with other file in this interface.

**Comment by kent-mcleod** [comment]
> anybody has an idea why `TX2_debug_gcc_64 FAILED` failes?

@lsf37 looks like the machine runner has lost permissions to the destination of the binary copy: 
`  Couldn't change mode of /tftpboot/tx2a/seL4-image: [Errno 1] Operation not permitted: '/tftpboot/tx2a/seL4-image'`

**Comment by lsf37** [comment]
> > anybody has an idea why `TX2_debug_gcc_64 FAILED` failes?
> 
> @lsf37 looks like the machine runner has lost permissions to the destination of the binary copy: ` Couldn't change mode of /tftpboot/tx2a/seL4-image: [Errno 1] Operation not permitted: '/tftpboot/tx2a/seL4-image'`

This bit is normal, and should really be fixed, it's happening for all runs (the user running this doesn't have enough permissions for changing permissions of the image, but doesn't really need to change them). The thing that went wrong in this one is

```
Retrieving file: /tx2a/pxelinux.cfg/default-arm
ethernet@2490000 Waiting for PHY auto negotiation to complete........ done
*** ERROR: `ipaddr' not set
 ```

which looks like a network fluke. I'll restart the test. @wom-bat this looks like a DHCP failure (but did work fine for all the other boards and also the other test runs of `tx2`).

**Comment by axel-h** [comment]
> Are you able to use `_` instead of `-` in the new filenames to be consistent with other file in this interface.

Changed to underscores.

**Review by lsf37 — APPROVED** [review]

**Comment by axel-h** [comment]
The changes affect ARM only, so I don't think the failing of HIFIVE_verification_gcc_64 is related.

**Review by kent-mcleod — APPROVED** [review]

**Comment by axel-h** [comment]
HW Run `IMX8MM_EVK, armv8a, gcc` and `IMX8MM_EVK, armv8a, clang` fail now, see the board is stuck?

**Comment by lsf37** [comment]
Yes, someone was holding the lock for that board for a few days and blocked all the tests (you can see who in the logs ;-)). Seems like nobody but me is policing this, so it breaks when I'm on leave..

**Pull request merged**